core/src/main/java/org/ehcache/core/statistics/CacheStatistics.java

```diff
@@ -41,6 +40,16 @@
    */
   Map<String, TierStatistics> getTierStatistics();
 
+  /**
+   * Register a derived statistic to one of the existing statistic.
+   *
+   * @param outcomeClass the enum of the possible outcomes
+   * @param statName name of the statistic we are looking for
+   * @param derivedStatistics derived statistic to register
+   * @param <T> type of the outcome
+   */
+  <T extends Enum<T>> void registerDerivedStatistics(Class<T> outcomeClass, String statName, ChainedOperationObserver<? super T> derivedStatistics);
```

Comment:
It would make calling code simpler if this was:
```
<T extends Enum<T>, S extends ChainedOperationObserver<? super T>> S registerDerivedStatistic(Class<T> type, String name, S statistic);
```

Reply:
Thanks

Reply:
Hum. It is actually worse... Nothing compiled with this syntax

Reply:
How did it not compile?

Reply:
I've changed it to void. Then, I had a compilation issue.  But just realized I was being stupid. You remove the s at the end. So that's the only reason... Doh. Will push the change.